misc/config_tools/static_allocators/gpa.py

```diff
@@ -109,15 +109,22 @@ def overlaps(self, other):
             return False
         return True
 
-def insert_vuart_to_dev_dict(scenario_etree, devdict_32bits):
+def insert_vuart_to_dev_dict(scenario_etree, vm_id, devdict_32bits):
+
     console_vuart =  scenario_etree.xpath(f"./console_vuart[base != 'INVALID_PCI_BASE']/@id")
-    communication_vuarts = scenario_etree.xpath(f".//communication_vuart[base != 'INVALID_PCI_BASE']/@id")
     for vuart_id in console_vuart:
         devdict_32bits[(f"{VUART}_{vuart_id}", "bar0")] = PCI_VUART_VBAR0_SIZE
         devdict_32bits[(f"{VUART}_{vuart_id}", "bar1")] = PCI_VUART_VBAR1_SIZE
-    for vuart_id in communication_vuarts:
-        devdict_32bits[(f"{VUART}_{vuart_id}", "bar0")] = PCI_VUART_VBAR0_SIZE
-        devdict_32bits[(f"{VUART}_{vuart_id}", "bar1")] = PCI_VUART_VBAR1_SIZE
+
+    vm_name = common.get_node(f"//vm[@id = '{vm_id}']/name/text()", scenario_etree)
+    communication_vuarts = scenario_etree.xpath(f"//vuart_connection[endpoint/vm_name/text() = '{vm_name}']")
+    vuart_id = 1
+    for vuart in communication_vuarts:
```

Comment:
You can use `for vuart_id, vuart in enumerate(communication_vuarts, start=1)` to simplify the code.

Reply:
OK